Extension/src/Debugger/configurationProvider.ts

```diff
@@ -262,126 +372,37 @@ export class CppConfigurationProvider implements vscode.DebugConfigurationProvid
-
-            // If pipeProgram does not get replaced and there is a pipeCwd, concatenate with pipeProgramStr and attempt to replace.
-            if (!replacedPipeProgram && !path.isAbsolute(pipeProgramStr) && config.pipeTransport.pipeCwd) {
-                const pipeCwdStr: string = config.pipeTransport.pipeCwd.toLowerCase().trim();
-                const newPipeProgramStr: string = path.join(pipeCwdStr, pipeProgramStr);
-
-                replacedPipeProgram = debugUtils.ArchitectureReplacer.checkAndReplaceWSLPipeProgram(newPipeProgramStr, debugUtils.ArchType.ia32);
-            }
-
-            if (replacedPipeProgram) {
-                config.pipeTransport.pipeProgram = replacedPipeProgram;
-            }
-        }
-
-        const macOSMIMode: string = config.osx?.MIMode ?? config.MIMode;
-        const macOSMIDebuggerPath: string = config.osx?.miDebuggerPath ?? config.miDebuggerPath;
-
-        const lldb_mi_10_x_path: string = path.join(util.extensionPath, "debugAdapters", "lldb-mi", "bin", "lldb-mi");
+    public static get recentBuildTaskLableStr(): string {
+        return DebugConfigurationProvider.recentBuildTaskLable;
+    }
 
-        // Validate LLDB-MI
-        if (os.platform() === 'darwin' && // Check for macOS
-            fs.existsSync(lldb_mi_10_x_path) && // lldb-mi 10.x exists
-            (!macOSMIMode || macOSMIMode === 'lldb') &&
-            !macOSMIDebuggerPath // User did not provide custom lldb-mi
-        ) {
-            const frameworkPath: string | undefined = this.getLLDBFrameworkPath();
+    public static set recentBuildTaskLableStr(recentTask: string) {
+        DebugConfigurationProvider.recentBuildTaskLable = recentTask;
+    }
 
-            if (!frameworkPath) {
-                const moreInfoButton: string = localize("lldb.framework.install.xcode", "More Info");
-                const LLDBFrameworkMissingMessage: string = localize("lldb.framework.not.found", "Unable to locate 'LLDB.framework' for lldb-mi. Please install XCode or XCode Command Line Tools.");
+    private buildAndDebugActiveFileStr(): string {
+        return `${localize("build.and.debug.active.file", 'build and debug active file')}`;
+    }
 
-                vscode.window.showErrorMessage(LLDBFrameworkMissingMessage, moreInfoButton)
-                    .then(value => {
-                        if (value === moreInfoButton) {
-                            const helpURL: string = "https://aka.ms/vscode-cpptools/LLDBFrameworkNotFound";
-                            vscode.env.openExternal(vscode.Uri.parse(helpURL));
-                        }
-                    });
+    private isClConfiguration(configurationLabel: string): boolean {
+        return configurationLabel.startsWith("C/C++: cl.exe") ? true : false;
+    }
 
-                return undefined;
+    private showErrorClNotAvailable(configurationLabel: string): boolean {
+        if (this.isClConfiguration(configurationLabel)) {
```

Comment:
everywhere that I see the `showError`, there is already a call to `isClConfiguration`.

My recommendation to avoid the duplicate call is to just do it here, add "if" to the function name (mentioned in another comment), and return false instead of throwing an Error if the config is not for CL.